### PR TITLE
Remove unused dependencies

### PR DESCRIPTION
**Pull request by neworld**
I was just curious, why do you need log4j and try to delete dependencies. Everything works just fine. Is it not a big deal, just good hygiene.

### CODE REVIEW

**Comment by quittle** [comment]
I am using it for IOUtils and NullByteStream so I'm surprised this builds. I'll check this out later today and see what's going on. Better to rip out minor deps like this if I can, I suppose.

**Comment by neworld** [comment]
> I am using it for IOUtils and NullByteStream so I'm surprised this builds. I'll check this out later today and see what's going on. Better to rip out minor deps like this if I can, I suppose.

Friendly ping. Did you have an opportunity to check? Because there is another vulnerability with log4j and needs a new patch: https://www.lunasec.io/docs/blog/log4j-zero-day-update-on-cve-2021-45046/

BTW, if log4j is not used, then the attack surface is much smaller.

**Comment by quittle** [comment]
TL;DR; I have removed the dependency on `org.apache.logging.log4j:log4j-iostream` and published `0.4.3` of this plugin without it. **Note** This again was published without any integration tests so I can't verify that the actual plugin works as expected, though I would really expect a compiler error for this change to validate if it completely broke.

---

Hey, sorry about the delay; things are busy for me with the holidays. I have been investigating what's going on with my build and it seems I can reasonably remove my dependency on log4j but should keep the Apache Commons IO one since the plugin does have direct dependencies on it.

You can still remove it from build.gradle and get away with it because the dependency can transitively be provided by the android gradle plugin (as a regular java library). This is not ideal however and breaks if you upgrade that dependency. I believe there's some magic to isolate the plugins because otherwise, you couldn't use a newer version of the Android plugin alongside this.

I really need to fix this project's ability to run tests to validate the plugin actually works outside of unit tests. Unfortunately, this is a side-project I no longer use and keep support for just for the community.

**Comment by neworld** [comment]
> You can still remove it from build.gradle and get away with it because the dependency can transitively be provided by the android gradle plugin (as a regular java library). This is not ideal however and breaks if you upgrade that dependency. I believe there's some magic to isolate the plugins because otherwise, you couldn't use a newer version of the Android plugin alongside this.

Totally make sense to me

> Unfortunately, this is a side-project I no longer use and keep support for just for the community.

We could help you with the burden. I'm just curious, maybe do you find another tool/way to deal with a similar problem instead of this plugin?

**Comment by quittle** [comment]
The original reason I built this was to enable easy running in CI and so I could just run a single gradle command locally and run the full suite of tests without having to set up the emulator separately, the same reason I created my [Setup Android SDK](https://github.com/quittle/gradle-setup-android-sdk) plugin.

There were a couple of reasons I stopped relying on it.
1. I was unable to find a free CI system I could use it with because none of them support a hypervisor for running the same architecture, running a different architecture was *super* slow on the low perf machines you get for free so tests would timeout and have flakiness in addition to adding hours to a build. This meant it wasn't useful for consumption and was difficult to test.
2. I switched machines from Linux to Windows and found the plugin didn't always work as expected (though I can't remember the details now), so I just ran everything else via command line and ran the instrumentation tests via Android Studio.
3. I slowed down my personal Android development, though still do some occasionally, so this has been less of a priority for me.

**Comment by quittle** [comment]
I just took another look at Circle CI and it seems they finally support nested virtualization so if I can get this running on there easily, I may be much more comfortable iterating and releasing this project going forward

**Comment by neworld** [comment]
We are running on KVM-enabled docker containers on Linux (inside our infrastructure). The plugin works very well, no single incident even we are running hundreds of builds each day

**Comment by quittle** [comment]
🤯 Woah, that's awesome! I had no idea anyone had taken a serious dependency on it. It's really hard to get a sense of that from just GitHub stars and maven downloads.

As I said before, I'm not actively working on this project but if you are interested in or need a feature, I'd be happy to discuss it and help you land a PR for it.
